### ISSUE TITLE
A valid ReactComponent must be returned.

### ISSUE THREAD

**Issue by mskeving** [codesuki/react-d3-components#82]
I may be missing something, but I can't seem to get a component up and running with the current setup I have. 

Here's what it looks like, based on the example provided to create a LineChart:

```
React   = require('react')
ReactD3 = require('react-d3-components')

LineChart = React.createFactory(ReactD3.LineChart)

module.exports = React.createClass
  displayName: 'LineChartDemo'

  render: ->
    data = [
        {
        label: 'somethingA',
        values: [{x: 0, y: 2}, {x: 1.3, y: 5}, {x: 3, y: 6}, {x: 3.5, y: 6.5}, {x: 4, y: 6}, {x: 4.5, y: 6}, {x: 5, y: 7}, {x: 5.5, y: 8}]
        },
        {
        label: 'somethingB',
        values: [{x: 0, y: 3}, {x: 1.3, y: 4}, {x: 3, y: 7}, {x: 3.5, y: 8}, {x: 4, y: 7}, {x: 4.5, y: 7}, {x: 5, y: 7.8}, {x: 5.5, y: 9}]
        }
    ]

    return LineChart
      data: data
      width: 400
      height: 400
      margin: {top: 10, bottom: 50, left: 50, right: 10}
```

But I can't seem to get a valid ReactComponent out of this. I get this error:
`Uncaught Error: Invariant Violation: LineChart.render(): A valid ReactComponent must be returned. You may have returned undefined, an array or some other invalid object.`

Is this an issue others have run into? Or maybe I'm missing a step?

**Comment by mskeving** [comment]
Actually, I seem to have figured out what was going on... It looks like it didn't have anything to do with CoffeeScript, or not creating a component correctly. It had to do with a discrepancy of React versions.

For my application I am using React 0.13.3. react-d3-components has React listed as a peerDependency with version 0.14.0. So when I installed this, a new version of React was pulled in along with it, which has no knowledge of the components the global react has created. So when I tried using your components, I kept getting that error message:

`Uncaught Error: Invariant Violation: LineChart.render(): A valid ReactComponent must be returned. You may have returned undefined, an array or some other invalid object.`

It's a pretty cryptic message given the cause of it, but as soon as I removed /react-d3-components/node_modules/react/, the LineChart displayed perfectly.

Is there a reason react 0.14.0 was listed? It seems to work fine with this earlier version. Might be worth listing a range of acceptable react versions instead of just one?

**Comment by codesuki** [comment]
Good idea and sorry for the trouble! I am sorry I must have missed this issue during christmas holiday >_<

**Comment by codesuki** [comment]
The reason react 0.14.0 was listed is because of some attribute that is used in line charts and only available from 0.14.

**Comment by mskeving** [comment]
Okay - no worries! I figured everything out (mostly just learned how peerDepencies work) so it wasn't really an issue anymore. Might be worth putting a note somewhere in case someone else runs into it as it took quite a while to figure out what was going on.
